src/main/java/com/google/sps/servlets/EventServlet.java

```diff
@@ -239,6 +239,8 @@ private String addEvent(HttpServletRequest request, HttpServletResponse response
 
     Event event = eventBuilder.build();
 
+    String eventId = eventStorageObject.addOrUpdateEvent(event);
+    event.setId(eventId);
```

Comment:
You only need to call addOrUpdate twice here if you need to create the calendar event. 

Nitpicking: It is kind of hacky to call it twice because you store the id in the calendar entry. You could solve this by just having an index on the calendar id for the events in the datastore if you want to find the events by gcalendar id. But it is fine for now :)